### DISCUSSION TITLE
Feature: Custom Shapes

### DISCUSSION THREAD

**Discussion by doctorlard** [tone-row/flowchart-fun#29] [Features / Syntax]
I love it :-) How about a sentence ending in a question mark (?) could be drawn as a diamond, with arrows coming from the corners (for the first three branches anyway). e.g.
```
Is it morning yet?
  yes: get up
  no: go back to sleep
    wake up: (1)
```

**Comment by rob-gordon** [comment]
That's an interesting idea. Someone on hacker news mentioned that these are not actually flowcharts but rather directed graphs. This would help push things in the flowchart direction for sure.

 I'm migrating all issues related to updating/growing the syntax over to Github Discussions so I can gather feedback and gauge support.

Thanks for your suggestion!

**Comment by verazorita** [comment]
Having the possibility to chose the shape of the container is amazing. If I could have the same styles [Drakon](http://drakon-editor.sourceforge.net/) uses this would be a gamechanger.

**Comment by rob-gordon** [comment]
(moved from an issue) @rubjo:

> Very, very cool, thanks for making this!
> 
> Part of the appeal is its simplicity, of course, but: any plans to extend with a handful of different typical shapes (eg. oval, parallelogram, diamond)?

That's definitely on my radar, though as you said, I don't want to do it at the cost of its simplicity.

One thing that would really help me chart a course is a well-defined goal. For instance, does anyone know if there is a codified set of shapes for flowcharts? Is there even a codified definition for flowcharts? (Probably things I should have considered before building 😅 )  

If we can define a set of shapes to support then we'll be ready to start the conversation about the best syntax to support them.

**Comment by rob-gordon** [comment]
@doctorlard @veradrawer Just wanted to mention that custom shapes and colors are now possible using a kind of utility-class syntax.

```
[.star] I'm a star
[.star.bg-yellow] I'm a yellow star
[.color-blue] I have blue text
```

Hopefully this helps for some of your use-cases!

**The full list of shapes is here:** https://flowchart.fun/c#H4hQBsEME8HsFcAuAuUACNA7SBbApsmgOYBOAlgCagjCgDaAdHuOGQA4DOeAuvQ4uUiYi4HnxIJMFALQCyQkWMYk8AY0QLRvZZJkr1mpQwBGsRIlg5pE+FOtqNwrX1VJ7Bp0eOQSK8NoYSAAtLU0oAinkcWCkAmztI3BiKALY8TA0iGLjdaTSMyCzMAKC8AA9C7PFc0oqigNUY1UgANzxpWsrivlK2TKqdWxle-u7GWAN66qHpCdGAjg0SAMycmdW+NrEgA

**The full list of colors is here:**
https://flowchart.fun/c#H4hQBsEME8HsFcAuAuUACNA7SBbApsmgOYBOAlgCagjCgD0AVGgMKziwkDOaAxrPmgBmJfmkQALPCwDK0gIxppABzw9Q4xIiWdkdOhTwA3POxUkAdDlgAvMuCjmOROnkwBaAKrT9sHpzoA6ngARnTMsnR87CQA+oaQ4PB4oAx0oADa5lEcbsFQPADWALpoeZCFGVlsOZx2xiQlteD1ldkkbqQwJZ3QrdXtAO7iZIh4JUMjyZltbjiQIrCYJXMLmH3RbiR4FCVbVNP9bkrwJErgY2jHp+frOYLwPOK1kCX3j8+37aR4rt1brp83OAyPgSsD8IC2GRjCUocZAdATOwBiVEfZYANAdhDNAStjegcNnkkiViVMqhtRgkSlTwIDIABHeAvNCM5n0OiKXBSNrcSDxOyQPJSQQcMSSUrlAqkBCYfbmYJEXL5YqlFWVRVuJr1Rp1PAkDVKnp-GCGtwTUbjYajM0rWCLZbze1rTKava7bZmq5nC7em6upVvJ5kFlBj4Bjr-JbEKNm8EXeNmuEXZNmtHI1FIjFm-F4-kEhVKsmkxLkzW0ml4BJmtks2ugIA
